app/assets/javascripts/discourse/app/widgets/post-menu.js

```diff
@@ -728,12 +768,26 @@ export default createWidget("post-menu", {
     return contents;
   },
 
-  openAdminMenu() {
-    this.state.adminVisible = true;
-  },
-
-  closeAdminMenu() {
-    this.state.adminVisible = false;
+  openAdminMenu(event) {
+    this.menu.show(event.target, {
+      identifier: "admin-post-menu",
+      component: AdminPostMenu,
+      model: {
+        scheduleRerender: this.scheduleRerender,
+        transformedPost: this.attrs,
```

Comment:
Instead of repeating all these here and on line 514, why not just make a function that returns a JS object with all the options?

Reply:
no specific reason, I honestly do not care much about this part, it's not supposed to live long :)

Reply:
And I guess Im just happy it works correctly like this 😅